lib/test_track_rails_client/assignment_helper.rb

```diff
@@ -2,16 +2,18 @@ module TestTrackRailsClient::AssignmentHelper
   def stub_test_track_assignments(assignment_registry) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
     raise "Cannot stub test track assignments when TestTrack is enabled" if TestTrack.enabled?
 
-    split_registry = TestTrack::Fake::SplitRegistry.instance.to_h.dup
+    split_registry = TestTrack::Fake::SplitRegistry.instance.to_h.deep_dup
```

Comment:
was this the leakage?

Reply:
yeah, this was the leakage. the inner `split` hash was not duped, so modifications to it was persisted across tests.

i'll run it against a few apps to see if there is anything else going on.